### PR TITLE
Conda build update

### PR DESCRIPTION
**Pull request by kain88-de**
updates the conda build recipe, removes one last occurrence of tempdir and installs griddata from conda.

### CODE REVIEW

File: maintainer/conda/MDAnalysis/meta.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ test:
 
   commands:
     # run the testsuite with 2 processes
-    - python -c 'import MDAnalysisTests; MDAnalysisTests.run(label="full", extra_argv=["-v", "--processes=2", "--process-timeout=120"])'
+    - python -c 'import MDAnalysisTests; MDAnalysisTests.run(label="full", extra_argv=["-v"])'
```
Comment (jbarnoud) [inline maintainer/conda/MDAnalysis/meta.yaml]: Why did you remove the processes and the timeout flags?

Comment (kain88-de) [inline maintainer/conda/MDAnalysis/meta.yaml]: I only assign one core the the virtual machine that is used to build the package. So running two test processes makes no sense.

**Comment by orbeckst** [comment]
Any reason not to merge??

**Comment by kain88-de** [comment]
@MDAnalysis/coredevs does someone wants to review this?

**Comment by jbarnoud** [comment]
Is there a way to not install all the dependancies manually? Because of this, the installation of the package is only partially tested.

Except for that, it looks good to me.

**Comment by kain88-de** [comment]
> Is there a way to not install all the dependancies manually? Because of this, the installation of the package is only partially tested.

What? Do you mean that I list all the dependencies required for running and installing? There isn't really a way around this using any other package manager then PIP. pip is disallowed to install any packages manually (ie download from the web). This is to ensure that all dependencies are managed by conda. Chaos would ensue if every package installed it's own version of numpy.

**Comment by jbarnoud** [comment]
Would it be possible to install cython, numpy, and matplotlib—that are big stuff—with conda, and let pip take care of the rest?

**Comment by jbarnoud** [comment]
I mean in travis.

**Comment by kain88-de** [comment]
we could I just don't see what that would change. The same version of the package gets installed.

**Comment by jbarnoud** [comment]
On 22/05/16 23:23, kain88-de wrote:
>
> we could I just don't see what that would change. The same version of 
> the package gets installed.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/MDAnalysis/mdanalysis/pull/854#issuecomment-220857299>
>
Yes, but if we get to forget to update dependencies in setup.py we may 
very well not catch it with the current settings.

**Comment by kain88-de** [comment]
We can specify the exact version that conda should install. Otherwise pip and conda will pick the most up to date version. So we aren't checking older versions with pip either.

**Comment by jbarnoud** [comment]
On 22/05/16 23:35, kain88-de wrote:
>
> We can specify the exact version that conda should install. Otherwise 
> pip and conda will pick the most up to date version. So we aren't 
> checking older versions with pip either.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/MDAnalysis/mdanalysis/pull/854#issuecomment-220857881>
>
Is pip modified in anyway by conda? If not, then pip should install a 
version compatible with what specified in setup.py from pypi. It is the 
behavior we expect, and we should rely on it in the tests. If we do not 
rely on it, then we will not catch we it fails.

**Comment by kain88-de** [comment]
> Is pip modified in anyway by conda?

No



> If not, then pip should install a version compatible with what specified in setup.py from pypi.

Since we only specify minimum versions installing a compatible version means installing the newest release available on PyPi. We install only package from the official conda channel, which follows the PyPi releases closely. The GridDataFormats package is distributed by us so we know it is the latest release as well. pip would install exactly the same packages.

**Comment by jbarnoud** [comment]
I get that the same packages are installed by conda and by pip. My concern is that, by hardcoding the dependancies we install in `.travis.yaml`, we bypass `setup.py`; that means that we do not test that `setup.py` list all the requirements. Since we all have the requirements intalled already, and that we probably have installed requirements that could be added in the future, ww would probably not notice if something happened there.

**Comment by kain88-de** [comment]
> that means that we do not test that setup.py list all the requirements.

yes we do. If something isn't installed setup.py will just install it. 

> Since we all have the requirements intalled already, and that we probably have installed requirements that could be added in the future, ww would probably not notice if something happened there.

Are you worried that through the conda installation we do install packages right now that we don't depend on explicitly? If that is your concern. We are **already** doing that with pip. Both pip and conda as setup right now install a minimal number of packages to run MDAnalysis including all of our implicit dependencies (dependencies of our dependencies) that we don't require in our `setup.py`.

**Comment by jbarnoud** [comment]
On 23/05/16 16:10, kain88-de wrote:
>
>     that means that we do not test that setup.py list all the
>     requirements.
>
> yes we do. If something isn't installed setup.py will just install it.
>
>     Since we all have the requirements intalled already, and that we
>     probably have installed requirements that could be added in the
>     future, ww would probably not notice if something happened there.
>
> Are you worried that through the conda installation we do install 
> packages right now that we don't depend on explicitly? If that is your 
> concern. We are *already* doing that with pip. Both pip and conda as 
> setup right now install a minimal number of packages to run MDAnalysis 
> including all of our implicit dependencies (dependencies of our 
> dependencies) that we don't require in our |setup.py|.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/MDAnalysis/mdanalysis/pull/854#issuecomment-220991263>
>
I was triggered by the new dependencies that gets hard coded in that PR. 
But I had a look at the history, and indeed we rely on dependencies 
hardcoded in `.travis.yml` for a while. I guess it would be the object 
of an other PR.


I see you made a change not long ago, are you done? If so, it can be merged.

**Comment by kain88-de** [comment]
Yes I'm done. As explained the last change is to increase the build number as I made an error in the first package for the new release.
